helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml

```
@@ -65,19 +65,20 @@ spec:
         team: atlas
         topic: observability
         cancel_if_outside_working_hours: "true"
-
     - alert: PrometheusCriticalJobScrapingFailure
       annotations:
         description: {{`Prometheus {{$labels.installation}}/{{$labels.cluster_id}} has failed to scrape all targets in {{$labels.job}} job.`}}
         summary: Prometheus fails to scrape all targets in a job.
         opsrecipe: prometheus-job-scraping-failure/
-      expr: (count(up{app=~"kubernetes|kube-controller-manager|kube-scheduler|kubelet|node-exporter|kube-state-metrics"} == 0) BY (app,job, installation, cluster_id) / count(up{app=~"kubernetes|kube-controller-manager|kube-scheduler|kubelet|node-exporter|kube-state-metrics"}) BY (app, job, installation, cluster_id)) == 1
+      ## We ignore bastion hosts node exporters
+      expr: (count(up{app=~"kubernetes|kube-controller-manager|kube-scheduler|kubelet|node-exporter|kube-state-metrics", job!~".*bastions.*"} == 0) BY (app,job, installation, cluster_id) / count(up{app=~"kubernetes|kube-controller-manager|kube-scheduler|kubelet|node-exporter|kube-state-metrics", job!~".*bastions.*"}) BY (app, job, installation, cluster_id)) == 1
```

Now we have UTs, it was easy to try this thing I've been thinking about for a while:
```suggestion
      expr: |-
        (
          count(
            up{
              app=~"kubernetes|kube-controller-manager|kube-scheduler|kubelet|node-exporter|kube-state-metrics",
              job!~".*bastions.*"
            } == 0
          ) BY (app,job, installation, cluster_id)
          /
          count(
            up{
              app=~"kubernetes|kube-controller-manager|kube-scheduler|kubelet|node-exporter|kube-state-metrics",
              job!~".*bastions.*"
            }
          ) BY (app, job, installation, cluster_id)
        ) == 1
```

WDYT? Isn't it much easier to read and debug?

---

This looks amazing TBH